src/lib/components/asset.svelte

```diff
@@ -1,20 +1,22 @@
 <script lang="ts">
 	import Container from '$lib/components/container.svelte'
-	export let name = ''
-	export let token = ''
-	export let amount = 0
+	export let name: string
+	export let token: string
+	export let amount: bigint
+	export let decimals: number
+	export let image: string
 </script>
 
 <div class="asset root">
 	<Container justify="space-between" align="center" alignItems="center" direction="row">
 		<div class="token">
-			<img src="https://picsum.photos/48/48" alt={`${name} logo`} />
+			<img src={image} alt={`${name} logo`} />
 			<span class="text-lg text-bold">
 				{name}
 			</span>
 		</div>
 		<div class="text-lg">
-			{amount}
+			{(Number(amount) / 10 ** decimals).toFixed(2)}
```

Comment:
I'm not sure whether there can actually be a precision difference as the lower bits are all discarded, but if there is, this could potentially be more precise with:
```js
Number(amount / 10n ** BigInt(decimals - 2)) / 100
```